src/rustllvm/PassWrapper.cpp

```diff
@@ -299,10 +299,15 @@ static size_t getLongestEntryLength(ArrayRef<SubtargetFeatureKV> Table) {
 extern "C" void LLVMRustPrintTargetCPUs(LLVMTargetMachineRef TM) {
   const TargetMachine *Target = unwrap(TM);
   const MCSubtargetInfo *MCInfo = Target->getMCSubtargetInfo();
+  const Triple::ArchType HostArch = Triple(sys::getProcessTriple()).getArch();
+  const Triple::ArchType TargetArch = Target->getTargetTriple().getArch();
   const ArrayRef<SubtargetFeatureKV> CPUTable = MCInfo->getCPUTable();
   unsigned MaxCPULen = getLongestEntryLength(CPUTable);
 
   printf("Available CPUs for this target:\n");
+  if (HostArch == TargetArch)
+    printf("    %-*s - Select the CPU of the current host (currently %s).\n",
+      MaxCPULen, "native", sys::getHostCPUName().data());
```

Comment:
StringRefs are not guaranteed to be null terminated. Use `%*` and pass the length as well, as in https://github.com/rust-lang/rust/blob/5dfc84cfa72b405c194228b53c4de3f6474204ec/src/rustllvm/PassWrapper.cpp#L651-L655